docs/analyze.md

```diff
@@ -0,0 +1,28 @@
+## Conflict analysis
+
+We want to find a possible reason of conflict and add a clause that forbids this assignment. 
+In order to do this we look for a minimum set of literals which reason the conflict in implication graph. 
+
+### Implication graph
+
+Every time we make a decision to set a positive or negative value to variable,
+it causes new appointments to other variables of [unit propagation](propagation.md).
+
+Implication graph is the directed acyclic graph. It's a partial assignment of variables. 
+The decision literals don't have ancestors and added to the graph each on its own level. All
+literals derived by chosen literals and all previous assignments in the graph are added to 
+the current level. 
+
+### Conflict
+
+A pair of opposite literals is called a conflict. This means that 
```

Comment:
"a pair of opposite literals IN THE SAME CLAUSE is a conflict"

Reply:
Note that this clause might be "learnt"

Reply:
Technially, a "conflict" is a a complementary assignment to the already assigned variable. This also corresponds to "two different polarities of one variable on the trail" (however, a good solver does not actually place a conflicting value on the trail — it just immediately reports a conflict in this case).